network/irc/irc.go

```diff
@@ -541,6 +544,18 @@ func (bot *ircBot) sendPassword() {
 	bot.Send("NickServ", "identify "+bot.password)
 }
 
+func (bot *ircBot) sendSaslStart() {
+	bot.SendRaw("AUTHENTICATE PLAIN")
+	out := bytes.Join([][]byte{[]byte(bot.nick), []byte(bot.nick), []byte(bot.password)}, []byte{0})
+	encpass := base64.StdEncoding.EncodeToString(out)
+	bot.SendRaw("AUTHENTICATE " + encpass)
+	bot.SendRaw("AUTHENTICATE +")
+}
```

Comment:
Should work, but the 2nd `bot.SendRaw("AUTHENTICATE +")` seems unnecessary (PLAIN is a 1-step mechanism).

It would be better to split the burst into individual events:

 * receive `CAP … ACK` → send `AUTHENTICATE PLAIN`
 * receive `AUTHENTICATE …` → send `AUTHENTICATE <encpass>`